server/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -71,7 +73,37 @@ public TransportValidateQueryAction(Settings settings, ThreadPool threadPool, Cl
     @Override
     protected void doExecute(Task task, ValidateQueryRequest request, ActionListener<ValidateQueryResponse> listener) {
         request.nowInMillis = System.currentTimeMillis();
-        super.doExecute(task, request, listener);
+        LongSupplier timeProvider = () -> request.nowInMillis;
+        ActionListener<org.elasticsearch.index.query.QueryBuilder> rewriteListener = ActionListener.wrap(rewrittenQuery -> {
+            request.query(rewrittenQuery);
+            super.doExecute(task, request, listener);
+        }, ex -> {
+            List<QueryExplanation> explanations = new ArrayList<>();
```

Comment:
Soooo, @colings86 and I just discussed this and we arrived at what we feel is a better solution: use a new object, CoordinatingNodeRewriteError (or some TBD name) and add it as optional to the `ValidateQueryResponse`, which addresses the concerns I had about shoehorning this error into QueryExplanation which is really written to describe shard level operations.

Since this is a new, albeit optional thing, I'll also:

- [ ] Update documentation to reflect change and make it clear that this new explanation and List<QueryExplanation> are mutually exclusive in the response
- [ ] Check with Luca that high level REST client will serialize/de-serialize, I'll add code/tests for this if necessary
- [ ] include version checking in serialization to ensure backwards compatibility

Since Colin is on holiday next week, I'll annoy @jpountz for a review in his absence.

Reply:
If you don't mind, Colin is much more familiar than I am with this, can it wait?